packages/shared/src/guards.ts

```diff
@@ -388,3 +390,32 @@ export function isTypedNdArray<T extends NumArray>(
 ): ndArr is NdArray<Exclude<T, number[]>> {
   return ndArr.dtype !== 'array';
 }
+
+export function hasCompoundType<S extends Shape>(
+  dataset: Dataset<S>
+): dataset is Dataset<S, CompoundType> {
+  return dataset.type.class === DTypeClass.Compound;
+}
+
+export function assertCompoundType<S extends Shape>(
+  dataset: Dataset<S>
+): asserts dataset is Dataset<S, CompoundType> {
+  if (!hasCompoundType(dataset)) {
+    throw new Error('Expected dataset to have compound type');
+  }
+}
+
+export function hasPrintableFields<S extends Shape>(
```

Comment:
Although the current names make sense, it might be best to stick to the more consistent naming: `hasPrintableCompoundType` and `assertPrintableCompoundType`?